public/app/features/trails/DataTrail.tsx

```diff
@@ -223,6 +229,8 @@ export class DataTrail extends SceneObjectBase<DataTrailState> {
       stateUpdate.topScene = new MetricSelectScene({});
     }
 
+    stateUpdate.metricSearch = values.metricSearch?.toString() || undefined;
```

Comment:
This you need the same logic as above (only set this if it's a string or clear it if it's null )

Reply:
it's due to a strange behavior of url sync , values are only passed to this function if the URL values differ from state (so if they are the same but metric isis different then this will be called with values having metric value but not metricSearch). Something I have been wanting to change but not come up with a solution for yet https://github.com/grafana/scenes/pull/551